Firestore/tests/System/DocumentAndCollectionTest.php

```diff
@@ -44,6 +45,17 @@ public function testCreate()
         $this->assertEquals(['firstName' => 'Kate'], $document->snapshot()->data());
     }
 
+    public function testSnapshotWithReadTime()
+    {
+        $readTime = new Timestamp(new \DateTimeImmutable());
+        $snapshotData = $this->document->snapshot([
+            'readTime' => $readTime
+        ])->data();
```

Comment:
@yash30201 
I tested this out, and `$snapshotData` is returning `null`. This is using the project `php-docs-samples-kokoro1-fs`, but I do not think that should matter, as commenting out the `readTime` line gets rid of the issue.